docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+
+This endpoint can answer questions like:
+
+- Tell me all of the transactions for a given subsidy.
+- Within a subsidy, give me all transactions for a specific user or content key.
+- Given a user id and content key, tell me if a successful/committed transaction exists, that is,
+  has the user previously redeemed this subsidy for this content?
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which transactions should be listed.
+- ``include_aggregates`` (query param, optional): Specifies if aggregates (quantities, number of transactions) should be
+  returned as part of the paginated response.  Defaults to ``true``.
+- ``user_id`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``user_id`` value.
+- ``content_key`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``content_key`` value.
+
+Outputs
+-------
+Returns a paginated list of aggregate and transaction data:
+
+::
+
+   {
+       'previous': null,
+       'next': '...',
+       'count': 'blah',
+       'aggregates': {
+           'total_quantity': 12350,
+           'unit': 'USD_CENTS',
+           'remaining_balance': 987650
+       },
+       'results': [
+           {
+               'uuid': 'the-transaction-uuid',
+               'status': 'completed',  TODO: enumerate valid statuses
+               'idempotency_key': 'the-idempotency-key',
+               'user_id': 54321,
+               'content_key': 'demox_1234+2T2023',
+               'quantity': 19900,
+               'unit': 'USD_CENTS',
+               'reference_id': 1234,
+               'reference_table': 'enrollments',
+               'subsidy_access_policy_uuid': 'a-policy-uuid',
+               'metadata': {...},
+               'created': 'created-datetime',
+               'modified': 'modified-datetime',
+               'reversal': null
+           }
+       ]
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  The transaction data returned should be filtered to only those related to the requesting user, within
+  the subsidy specified in the request's URL path.
+
+enterprise_admin
+  Should return the requested subsidy transactions only if the requesting user has implicit (JWT) or explicit (DB-defined)
+  ``enterprise_admin`` role assigned for the requested subsidy's enterprise.
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to view **all**
+  subsidy records, and can therefore retrieve the requested subsidy transactions.
+
+GET (retrieve) enterprise subsidy transaction
+=============================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/[transaction-uuid]/**
+
+This endpoint can retrieve a single transaction, provided the caller has the subsidy and transaction uuid.
```

Comment:
Why do we need both the subsidy and transaction ID? Would it simplify this to just use the transaction ID?

Reply:
for the curation app, we opted to use `/api/v1/hightlight-sets` instead of `/api/v1/enterprise-curations/[enterprise_uuid]/highlight-sets`, presumably because the highlight set UUIDs were already globally unique.  In the case of subsidies and transactions, I think the situation is probably the same, so I'd tentatively agree with beggs here.

Reply:
that's a fair point.  will change!